File: Samples/Layout/Pages/VB.NET/Pages.vb
Imports System.IO
Imports BitMiracle.Docotic.Pdf.Layout

Namespace BitMiracle.Docotic.Pdf.Samples
    NotInheritable Class Pages
        Public Shared Sub Main()
            ' NOTE:
            ' When used in trial mode, the library imposes some restrictions.
            ' Please visit http://bitmiracle.com/pdf-library/trial-restrictions
            ' for more information.

            Const PathToFile As String = "Pages.pdf"
            PdfDocumentBuilder.Create().Generate(PathToFile,
                Sub(doc)
                    doc.Pages(
                        Sub(page)
                            page.Size(PdfPaperSize.A4, PdfPaperOrientation.Portrait).
                                MarginVertical(50).
                                MarginHorizontal(40).
                                BackgroundColor(New PdfGrayColor(95))

                            Dim watermarkColor = New PdfRgbColor(255, 0, 0)
                            page.Foreground().
                                Extend().
                                AlignCenter().
                                AlignMiddle().
                                TranslateX(50).
                                TranslateY(200).
                                Rotate(-55).
                                Text("Watermark").
                                Style(Function(t) t.Parent.FontSize(100).FontColor(watermarkColor, 50))

                            page.Header().
                                PaddingBottom(30).
                                AlignRight().
                                Text("Header")

                            Dim loremIpsum As String = File.ReadAllText("..\Sample Data\lorem-ipsum.txt")
                            page.Content().Text(loremIpsum)

                            page.Footer().
                                AlignCenter().
                                Text(Sub(t) t.CurrentPageNumber())
                        End Sub)

                    doc.Pages(
                        Sub(page)
                            page.Size(PdfPaperSize.A3, PdfPaperOrientation.Landscape).
                                MarginVertical(50).
                                ContentFromRightToLeft().
                                Content().
                                Text("Hello World")
                        End Sub)

                    doc.Pages(
                    Sub(page)
                        page.Size(100, 100).
                            Content().
                            Text("Custom page size")
                    End Sub)
                End Sub)

            Console.WriteLine($"The output is located in {Environment.CurrentDirectory}")
            Process.Start(New ProcessStartInfo(PathToFile) With {.UseShellExecute = True})
        End Sub
    End Class
End Namespace

File: Samples/Layout/Pages/C#/Pages.cs
using System;
using System.Diagnostics;
using System.IO;
using BitMiracle.Docotic.Pdf.Layout;

namespace BitMiracle.Docotic.Pdf.Samples
{
    static class Pages
    {
        static void Main()
        {
            // NOTE:
            // Without a license, the library won't allow you to create or read PDF documents.
            // To get a free time-limited license key, use the form on
            // https://bitmiracle.com/pdf-library/download

            LicenseManager.AddLicenseData("PUT-LICENSE-HERE");

            const string PathToFile = "Pages.pdf";
            PdfDocumentBuilder.Create().Generate(PathToFile, doc =>
            {
                doc.Pages(page =>
                {
                    page.Size(PdfPaperSize.A4, PdfPaperOrientation.Portrait)
                        .MarginVertical(50)
                        .MarginHorizontal(40)
                        .BackgroundColor(new PdfGrayColor(95));

                    PdfRgbColor watermarkColor = new(255, 0, 0);
                    page.Foreground()
                        .Extend()
                        .AlignCenter()
                        .AlignMiddle()
                        .TranslateX(50)
                        .TranslateY(200)
                        .Rotate(-55)
                        .Text("Watermark")
                        .Style(t => t.Parent.FontSize(100).FontColor(watermarkColor, 50));

                    page.Header()
                        .PaddingBottom(30)
                        .AlignRight()
                        .Text("Header");

                    string loremIpsum = File.ReadAllText(@"..\Sample Data\lorem-ipsum.txt");
                    page.Content().Text(loremIpsum);

                    page.Footer()
                        .AlignCenter()
                        .Text(t => t.CurrentPageNumber());
                });

                doc.Pages(page =>
                {
                    page.Size(PdfPaperSize.A3, PdfPaperOrientation.Landscape)
                        .MarginVertical(50)
                        .ContentFromRightToLeft()
                        .Content()
                        .Text("Hello World");
                });

                doc.Pages(page =>
                {
                    page.Size(100, 100)
                        .Content()
                        .Text("Custom page size");
                });
            });

            Console.WriteLine($"The output is located in {Environment.CurrentDirectory}");
            Process.Start(new ProcessStartInfo(PathToFile) { UseShellExecute = true });
        }
    }
}
